sds_data_manager/lambda_code/SDSCode/indexer.py

```diff
@@ -20,6 +19,176 @@
+    # TODO: add checks for SPICE or other
+    # data types
+    science_file = ScienceFile(filename)
+
+    try:
+        science_file = ScienceFile(filename)
+    except InvalidScienceFileError as e:
+        return http_response(status_code=400, body=str(e))
+
+    # setup a dictionary of metadata parameters to unpack in the
+    # file catalog table
+    metadata_params = science_file.get_file_metadata_params()
+    metadata_params["file_path"] = s3_filepath
+
+    if metadata_params["data_level"] != "l0":
+        return http_response(status_code=400, body="Invalid data level")
+
+    logger.info(f"Inserting {filename} into database")
+    # Add data to the file catalog and status tables
+    with Session(engine) as session:
+        session.add(models.FileCatalog(**metadata_params))
+        # Event time looks like this:
+        # "time": "2024-01-16T17:35:08Z"
+
+        # Parses the time string from the event to a datetime object.
+        ingestion_date = datetime.datetime.strptime(event["time"], "%Y-%m-%dT%H:%M:%SZ")
+        # Formats the datetime object to a string with the format "%Y%m%d".
+        ingestion_data_str = ingestion_date.strftime("%Y%m%d")
+        # Parses the newly formatted string back into a datetime
+        # object with the desired structure.
+        ingestion_date_object = datetime.datetime.strptime(ingestion_data_str, "%Y%m%d")
+
+        status_params = {
+            "file_to_create_path": s3_filepath,
+            "status": models.Status.SUCCEEDED,
+            "job_definition": None,
+            "ingestion_date": ingestion_date_object,
+        }
+        session.add(models.StatusTracking(**status_params))
+        session.commit()
+    return http_response(status_code=200, body="Success")
+
+
+def batch_event_handler(event, engine):
+    """Handler for Batch event
+
+    Parameters
+    ----------
+    event : dict
+        The JSON formatted document with the data required for the
+        lambda function to process
+    engine : sqlalchemy.engine.Engine
+        Engine object to interact with the database.
+
+    Returns
+    -------
+    dict
+        HTTP response
+    """
+
+    if event["detail-type"] != "Batch Job State Change":
+        return http_response(status_code=400, body="Invalid event")
+    if event["detail"]["status"] == "SUCCEEDED":
+        # TODO:
+        # Get filename and then
+        # write it to file catalog table
+        pass
+
+    # TODO: update status table
+
+    return http_response(status_code=200, body="Success")
+
+
+def custom_event_handler(event, engine):
+    """_summary_
+
+    Parameters
+    ----------
+    event : dict
+        The JSON formatted document with the data required for the
+        lambda function to process
+    engine : sqlalchemy.engine.Engine
+        Engine object to interact with the database.
+
+    Returns
+    -------
+    dict
+        HTTP response
+    """
+    if event["detail-type"] != "Job Started":
```

Comment:
It might be good to add a note at the top about what these events look like - eg what the valid values are and general structure for the batch files and this custom event handler.